### PR TITLE
Implement HTTP Return implicit status codes and content types

### LINKED ISSUE #491: Implement HTTP Return implicit status codes and content types
**Description:**
Refer to spec 2.3.5
https://github.com/ballerina-platform/module-ballerinax-azure.functions/blob/svc-impl/spec/spec.md#235-return-types

**Describe your problem(s)**

**Describe your solution(s)**

**Related Issues (optional):**

**Suggested Labels (optional):**

**Suggested Assignees (optional):**

### PR DESCRIPTION
**Pull request by Thevakumar-Luheerathan**
## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

File: native/src/main/java/io/ballerina/stdlib/azure/functions/FunctionCallback.java [hunk 1]
```diff
@@ -187,4 +157,134 @@ private boolean isContentTypeExist(BMap<BString , ?> headersMap) {
...
+                MapType mapContent = (MapType) tableContent.getConstrainedType();
+                if ("json".equals(mapContent.getConstrainedType().getName())) {
+                    headers.put(StringUtils.fromString(Constants.CONTENT_TYPE),
+                            StringUtils.fromString("application/json"));
+                }
+
+            }
+        } else if (result instanceof BDecimal || result instanceof Long || result instanceof Double ||
+                result instanceof Boolean) {
+            headers.put(StringUtils.fromString(Constants.CONTENT_TYPE), StringUtils.fromString("application/json"));
```
Comment (xlight05) [inline native/src/main/java/io/ballerina/stdlib/azure/functions/FunctionCallback.java:220]: We can extract this to constants

File: native/src/main/java/io/ballerina/stdlib/azure/functions/FunctionCallback.java [hunk 2]
```diff
@@ -187,4 +157,134 @@ private boolean isContentTypeExist(BMap<BString , ?> headersMap) {
...
+                    MapType mapContent = (MapType) tableContent.getConstrainedType();
+                    if ("json".equals(mapContent.getConstrainedType().getName())) {
+                        headers.put(StringUtils.fromString(Constants.CONTENT_TYPE),
+                                StringUtils.fromString("application/json"));
+                    }
+
+                }
+            }
+
+        } else if (result instanceof BTable) {
```
Comment (xlight05) [inline native/src/main/java/io/ballerina/stdlib/azure/functions/FunctionCallback.java:207]: Shall we add a test case for this?

File: native/src/main/java/io/ballerina/stdlib/azure/functions/FunctionCallback.java [hunk 3]
```diff
@@ -187,4 +157,134 @@ private boolean isContentTypeExist(BMap<BString , ?> headersMap) {
...
+        statusCode = StringUtils.fromString((String) statusCode);
+        respMap.put(StringUtils.fromString(Constants.STATUS_CODE), statusCode);
+    }
+
+    private void addContentTypeImplicitly(Object result, BMap headers) {
+        if (result instanceof BString) {
+            headers.put(StringUtils.fromString(Constants.CONTENT_TYPE), StringUtils.fromString("text/plain"));
+
+        } else if (result instanceof BXmlItem) {
+            headers.put(StringUtils.fromString(Constants.CONTENT_TYPE), StringUtils.fromString("application/xml"));
```
Comment (xlight05) [inline native/src/main/java/io/ballerina/stdlib/azure/functions/FunctionCallback.java:180]: Shall we add a test case for this?

File: native/src/main/java/io/ballerina/stdlib/azure/functions/FunctionCallback.java [hunk 4]
```diff
@@ -187,4 +157,134 @@ private boolean isContentTypeExist(BMap<BString , ?> headersMap) {
...
+            TableType tableContent = (TableType) tableResult.getType();
+            if ("map".equals(tableContent.getConstrainedType().getName())) {
+                MapType mapContent = (MapType) tableContent.getConstrainedType();
+                if ("json".equals(mapContent.getConstrainedType().getName())) {
+                    headers.put(StringUtils.fromString(Constants.CONTENT_TYPE),
+                            StringUtils.fromString("application/json"));
+                }
+
+            }
+        } else if (result instanceof BDecimal || result instanceof Long || result instanceof Double ||
```
Comment (xlight05) [inline native/src/main/java/io/ballerina/stdlib/azure/functions/FunctionCallback.java:218]: Shall we add a test case for this?

File: native/src/main/java/io/ballerina/stdlib/azure/functions/FunctionCallback.java [hunk 5]
```diff
@@ -187,4 +157,134 @@ private boolean isContentTypeExist(BMap<BString , ?> headersMap) {
...
+    }
+
+    private void addContentTypeImplicitly(Object result, BMap headers) {
+        if (result instanceof BString) {
+            headers.put(StringUtils.fromString(Constants.CONTENT_TYPE), StringUtils.fromString("text/plain"));
+
+        } else if (result instanceof BXmlItem) {
+            headers.put(StringUtils.fromString(Constants.CONTENT_TYPE), StringUtils.fromString("application/xml"));
+
+        } else if (result instanceof BArray) {
```
Comment (xlight05) [inline native/src/main/java/io/ballerina/stdlib/azure/functions/FunctionCallback.java:182]: Shall we add a test case for this?

File: ballerina-tests/main.bal [hunk 6]
```diff
@@ -49,6 +49,14 @@ service /hello on ep {
         return err;
     }
 
+    resource function post nonHttpResTest1(@af:Payload string greeting) returns string {
+        string s1 = "alpha";
+        return s1;
+    }
+    resource function post nonReturnTest1(@af:Payload string greeting) {
```
Comment (anuruddhal) [inline ballerina-tests/main.bal:56]: Add a new line

File: native/src/main/java/io/ballerina/stdlib/azure/functions/FunctionCallback.java [hunk 7]
```diff
@@ -73,86 +83,46 @@ public void notifySuccess(Object result) {
...
-                Constants.COSMOS_DBOUTPUT.equals(outputBinding)) {
+        if (result == null) {
+            BString statusCode = StringUtils.fromString("202");
+            BMap<BString, Object> respMap =
+                    ValueCreator.createMapValue(TypeCreator.createMapType(PredefinedTypes.TYPE_ANYDATA));
+            respMap.put(StringUtils.fromString(Constants.STATUS_CODE), statusCode);
+            mapValue.put(StringUtils.fromString(Constants.RESP), respMap);
+            future.complete(mapValue);
+            return;
+
```
Comment (anuruddhal) [inline native/src/main/java/io/ballerina/stdlib/azure/functions/FunctionCallback.java:97]: Remove extra line

**Review by anuruddhal — CHANGES_REQUESTED** [review]

**Review by xlight05 — APPROVED** [review]

**Pull request merged**